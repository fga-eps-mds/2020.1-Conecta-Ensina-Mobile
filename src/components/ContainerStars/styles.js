import styled from 'styled-components';

export const Container = styled.TouchableOpacity`
  background-color: ${(props) => props.theme.colors.cinzaClaro};
  width: 95px;
  height: 95px;
  border-radius: 10px;
  margin-right: 20px;
  margin-bottom: 20px;
  align-items: center;
`;

export const Imagem = styled.Image`
  margin-top: 6px;
  height: 48px;
  width: 40px;
`;

export const Nome = styled.Text`
  padding-top: 3px;
  font-size: 12px;
`;
