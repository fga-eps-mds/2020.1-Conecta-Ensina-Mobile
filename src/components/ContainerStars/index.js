import React from 'react';

import {Container, Imagem, ContainerTexto} from './styles';
import {Rating} from 'react-native-ratings';
import CustomText from '../CustomText';

export default function SquareButton({data}) {
  return (
    <Container>
      <Imagem source={data.img} />
      <Rating
        type="custom"
        ratingCount={5}
        imageSize={9}
        style={{paddingTop: 8}}
        tintColor={'#D9D9D9'}
      />
      <ContainerTexto>
        <CustomText smaller black>
          {data.nome}
        </CustomText>
      </ContainerTexto>
    </Container>
  );
}
