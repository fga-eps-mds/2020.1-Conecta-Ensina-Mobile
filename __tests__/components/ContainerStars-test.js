import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import ContainerStars from '../../src/components/ContainerStars';

describe('Testing ContainerStars component', () => {
  const data = {
    firstName: 'name',
  };

  test('Should render component', () => {
    const tree = render(<ContainerStars data={data} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
