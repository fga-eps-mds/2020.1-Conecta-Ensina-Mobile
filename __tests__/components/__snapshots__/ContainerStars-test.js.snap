// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ContainerStars component Should render component 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D9D9D9",
      "borderRadius": 10,
      "height": 95,
      "marginBottom": 20,
      "marginRight": 20,
      "opacity": 1,
      "width": 95,
    }
  }
>
  <Image
    style={
      Array [
        Object {
          "height": 48,
          "marginTop": 6,
          "width": 40,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "fontSize": 12,
          "paddingTop": 3,
        },
      ]
    }
  >
    <Text
      black={true}
      smaller={true}
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "Roboto",
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      name
    </Text>
  </View>
</View>
`;
